src/Dialect/ONNX/ONNXOps/Math/ElementwiseUnary.cpp

```diff
@@ -154,7 +154,8 @@ LogicalResult ONNXCastLikeOp::inferShapes(
   if (!hasShapeAndRank(getInput()))
     return success();
 
-  Type elementType = (*this)->getAttr("to").cast<::TypeAttr>().getValue();
+  Type elementType =
+      getTargetType().getType().dyn_cast<RankedTensorType>().getElementType();
```

Comment:
getElementType() from TypeUtilities is convenient here: `getElementType(getTargetType().getType())` 

otherwise I would nit pick about dyn_cast vs cast and RankedTensorType vs ShapedType :)

Reply:
Funny enough... I think you still nit picked :p